test/FunctionPointer/lifetime.cpp

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2015 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <stdio.h>
+#include "lifetime.hpp"
+#include "core-util/FunctionPointer.h"
+
+class LifetimeChecker {
+public:
+    LifetimeChecker(int arg) : _arg(arg) {
+        _activeInstances++;
+    }
+    LifetimeChecker(const LifetimeChecker & lc) : _arg(lc._arg) {
```

Comment:
should probably also define a copy-assignment operator, in case a future implementation uses that (https://en.wikipedia.org/wiki/Rule_of_three_(C%2B%2B_programming)#Rule_of_Three)